schema_watcher.go

```diff
@@ -568,11 +568,11 @@ func (s *SchemaWatcher) getFileDescriptorSet(ctx context.Context) (*descriptorpb
 		// try to fallback to cache
 		data, cacheErr := s.cache.Load(ctx, s.cacheKey)
 		if cacheErr != nil {
-			return nil, "", time.Time{}, fmt.Errorf("%w (failed to load from cache: %v)", err, cacheErr)
+			return nil, "", time.Time{}, fmt.Errorf("%w (failed to load from cache: %w)", err, cacheErr)
```

Comment:
You can move these two lines or this function, etc, out to a file with a `go1.20` build tag, and preserve the old behavior for `!go1.20`.

That way we don't have to bump the required version in `go.mod`.

From recent discussion, we want to support latest three versions in our OSS stuff:
https://github.com/bufbuild/skeleton-oss-go/pull/3#issuecomment-1679629076